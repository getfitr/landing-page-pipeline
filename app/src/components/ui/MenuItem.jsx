import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from "prop-types";
import { Text } from '@chakra-ui/layout';

const MenuItem = ({ children, isLast, linkClassName, to, ...rest }) => (
  <Text
    mb={{ base: isLast ? 0 : 8, sm: 0 }}
    mr={{ base: 0, sm: isLast ? 0 : 8 }}
    display="block"
    {...rest}
  >
    <Link to={to} className={linkClassName}>{children}</Link>
  </Text>
);

MenuItem.propTypes = {
  isLast: PropTypes.bool,
  to: PropTypes.string,
  linkClassName: PropTypes.string,
};

MenuItem.defaultProps = {
  children: <div>Default menu item</div>,
  isLast: false,
  linkClassName: '',
  to: "/",
};

export default MenuItem;
