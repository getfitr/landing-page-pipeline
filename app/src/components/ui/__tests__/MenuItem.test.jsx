import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MenuItem from '../MenuItem';
import { Button } from '@chakra-ui/button';

test('Render a simple menu item', () => {
  const item = create(
    <BrowserRouter>
      <MenuItem />
    </BrowserRouter>
  );
  const tree = item.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render a menu item with text', () => {
  const item = create(
    <BrowserRouter>
      <MenuItem to="/test">A Menu Item</MenuItem>
    </BrowserRouter>
  );
  const tree = item.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render a menu button', () => {
  const item = create(
    <BrowserRouter>
      <MenuItem to="/button" isLast>
        <Button>A Button</Button>
      </MenuItem>
    </BrowserRouter>
  );
  const tree = item.toJSON();
  expect(tree).toMatchSnapshot();
});
