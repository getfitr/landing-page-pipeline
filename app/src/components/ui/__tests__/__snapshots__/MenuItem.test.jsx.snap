// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a menu button 1`] = `
<p
  className="chakra-text css-1g5pnks"
>
  <a
    className=""
    href="/button"
    onClick={[Function]}
  >
    <button
      className="chakra-button css-oqfhsh"
      type="button"
    >
      A Button
    </button>
  </a>
</p>
`;

exports[`Render a menu item with text 1`] = `
<p
  className="chakra-text css-n7rqg5"
>
  <a
    className=""
    href="/test"
    onClick={[Function]}
  >
    A Menu Item
  </a>
</p>
`;

exports[`Render a simple menu item 1`] = `
<p
  className="chakra-text css-n7rqg5"
>
  <a
    className=""
    href="/"
    onClick={[Function]}
  >
    <div>
      Default menu item
    </div>
  </a>
</p>
`;
